cmd/juju/model/defaultscommand.go

```diff
@@ -45,22 +51,46 @@ See also:
 
 // NewDefaultsCommand wraps defaultsCommand with sane model settings.
 func NewDefaultsCommand() cmd.Command {
-	return modelcmd.WrapController(&defaultsCommand{})
+	defaultsCmd := &defaultsCommand{
+		newCloudAPI: func(caller base.APICallCloser) cloudAPI {
+			return cloudapi.NewClient(caller)
+		},
+		newDefaultsAPI: func(caller base.APICallCloser) defaultsCommandAPI {
+			return modelmanager.NewClient(caller)
+		},
+	}
+	defaultsCmd.newAPIRoot = defaultsCmd.NewAPIRoot
+	return modelcmd.WrapController(defaultsCmd)
 }
 
 // defaultsCommand is compound command for accessing and setting attributes
 // related to default model configuration.
 type defaultsCommand struct {
-	modelcmd.ControllerCommandBase
-	api defaultsCommandAPI
 	out cmd.Output
+	modelcmd.ControllerCommandBase
 
-	action func(defaultsCommandAPI, *cmd.Context) error // The function handling the input, set in Init.
-	keys   []string
-	reset  bool // Flag indicating if we are resetting the keys provided.
+	newAPIRoot     func() (api.Connection, error)
+	newDefaultsAPI func(base.APICallCloser) defaultsCommandAPI
+	newCloudAPI    func(base.APICallCloser) cloudAPI
+
+	action                func(defaultsCommandAPI, *cmd.Context) error // The function handling the input, set in Init.
+	key                   string
+	resetKeys             []string
+	cloudName, regionName string
+	// reset holds the contents of the reset flag. It contains a list of comma
```

Comment:
I'll modify it. I am fine with removing many of the comments. I mostly wrote them out to rubber duck the logistics.